### ISSUE TITLE
newsboat doesn't cache read/unread between two computers using dropbox

### ISSUE THREAD

**Issue by evantravers** [newsboat/newsboat#884]
**Newsboat version (copy the output of `newsboat -v` or the first line of `git show`)**:
```
newsboat 2.19.0 - https://newsboat.org/
Copyright (C) 2006-2015 Andreas Krennmair
Copyright (C) 2015-2020 Alexander Batischev
Copyright (C) 2006-2017 Newsbeuter contributors
Copyright (C) 2017-2020 Newsboat contributors

Newsboat is free software licensed under the MIT License. (Type `newsboat -vv' to see the full text.)
It bundles JSON for Modern C++ library, licensed under the MIT License: https://github.com/nlohmann/json

newsboat 2.19.0
System: Darwin 19.4.0 (x86_64)
Compiler: g++ 4.2.1 Compatible Apple LLVM 11.0.0 (clang-1100.0.33.17)
ncurses: ncurses 5.7.20081102 (compiled with 5.7)
libcurl: libcurl/7.64.1 (SecureTransport) LibreSSL/2.8.3 zlib/1.2.11 nghttp2/1.39.2 (compiled with 7.64.1)
SQLite: 3.31.1 (compiled with 3.31.1)
libxml2: compiled with 2.9.4
```

Config file (copy from ~/.newsboat/config or ~/.config/newsboat/config):

```
# general settings
auto-reload yes
max-items 500
show-read-feeds no
text-width 50
reload-threads 11

# set browser for osx
browser /usr/bin/open -a "/Applications/Brave Browser.app" '%u'

# style
articlelist-format "%4i  %f  %D  %-20a  %-60t  %-40T"
datetime-format "%Y-%m-%d"

# unbind keys
unbind-key ENTER
unbind-key j
unbind-key k
unbind-key J
unbind-key K
unbind-key ^D
unbind-key ^U
unbind-key o
unbind-key g
unbind-key G

# bind keys - vim style
bind-key j down
bind-key k up
bind-key l open
bind-key h quit
bind-key ^D pagedown
bind-key ^U pageup
bind-key b toggle-source-view
bind-key U toggle-show-read-feeds
bind-key u show-urls
bind-key g home
bind-key G end
bind-key b open-in-browser
bind-key i sort
bind-key I rev-sort

# solarized
color background         default   default
color listnormal         color59   default
color listnormal_unread  default   default bold
color listfocus          black     cyan
color listfocus_unread   black     cyan bold
color info               default   black
color article            default   default

# highlights
highlight article "^(Title):.*$" blue default
highlight article "https?://[^ ]+" red default
highlight article "\\[image\\ [0-9]+\\]" green default
```

Steps to reproduce the issue:

1. I've symlinked `~/Dropbox/Sync/newsboat` to `~/.config/newsboat` on both computers. (`ln -s ~/Dropbox/Sync/newsboat/ ~/.config/newsboat`

2. I open one computer, use newsboat, close it. Dropbox syncs.

3. Open it on the other computer… I expect to see the read/unread cache'd between computers because the cache file is sync'd by dropbox, but it doesn't "remember". Could it be a relative/absolute path thing? Maybe I don't understand how it keeps track of read/unread.

One computer is my personal, one is my work… they have different usernames.

**Comment by Minoru** [comment]
Hi! Read/unread status is kept in the cache file under ~/.local/share/newsboat ([docs on XDG support](https://newsboat.org/releases/2.19/docs/newsboat.html#_xdg_base_directory_support)). You have to sync that too.

**Comment by evantravers** [comment]
Thank you for the quick response @Minoru, that's very kind of you.

If I'm understanding you correctly, I'll either need to just use ~/.newsboat so that I can use both… when I moved from using ~/.newsboat to ~/.config/newsboat I "opted in" to needing the ~/.local/share/newsboat folder too… I may move back to ~/.newsboat for simplicity of syncing.

Thanks again!

**Comment by Minoru** [comment]
> when I moved from using ~/.newsboat to ~/.config/newsboat I "opted in" to needing the ~/.local/share/newsboat folder too

Yup, exactly! It's either XDG directories or dotdir, no mixing.

**Comment by evantravers** [comment]
For anyone who ends up googling newsboat and dropbox, this was the solution. For simplicity, I moved back to dotdir, but you could also symlink your XDG directories. I was confused because I had started with the dotdir, `mv`'d the folder to `~/.config/newsboat/`, and newsboat had created _another_ `cache.db` file at `~/.local/share/newboat` that I didn't know about.

I was syncing the cache, just not the right one. I'm all good now!

**Comment by s1011205** [comment]
Hello, Evan Travers! I am wanting to have newsboat synced between two Mac computers with dropbox, but I am confused with the talk of XDG directories and dotdir, and I am not sure what your April 13, 2020 post means. I have had both a ~/.newsboat folder with a cache file, a urls file, and config file in it, and a ~/.local/share/newsboat folder with a cache file in it. So do I create two folders in Dropbox (one for each of those folders), put the files from the two original files in those folder and then create two symlinks with the terminal ln command?

**Comment by evantravers** [comment]
`newsboat` needs to know where it's configuration files, and it's .cache is (your read/unread).

Traditionally, unix executables used a "dotdir," something like `~/.newsboat`. The new spec is to use XDG directory which is usually set to `~/.config/newsboat/`. `newsboat` will check either and both, it's just a preference as to how you like to configure your files.

If you put your files and cache in dropbox, and symlink the folder to _either_ of those locations, you should be golden. In my case, my config was in dotdir, and my store was in XDG, and that confused me… but that means that hopefully I can help you.

**Comment by s1011205** [comment]
Hey, Evan! Thank you for your response. I tried to do the two following commands in terminal:
ln -s /Users/Paul/Dropbox/09Newsboat/home-dotlocal-share-newsboat /Users/Paul/.local/share/newsboat
ln -s /Users/Paul/Dropbox/09Newsboat/home-dotnewsboat /Users/Paul/.newsboat

This created a 'home-dotnewsboat' symlink folder in the /Users/Paul/.newsboat folder, and it created a 'home-dotlocal-share-newsboat' Symlink folder in the /Users/Paul/.local/share/newsboat local folder. I had the config file, cache file, and urls file in the 'home-dotnewsboat' folder in Dropbox, and the read/unread tracker cache file in the 'home-dotlocal-share-newsboat' folder in Dropbox.

After I did that, I tried to start Newsboat in the terminal, and I received the 'Error: no URLs configured. Please fill the file /Users/Paul/.newsboat/urls with RSS feed URLs or import an OPML file.' So apparently, it did not follow the symlink to the urls file in the 'home-dotnewsboat' Dropbox folder. It did create a new cache folder in the /Users/Paul/.newsboat folder on my local iMac. So it seems that my symlink.

What did I do wrong? Thank you for your help, Evan. I noticed in your post from 2020 that you had "For anyone who ends up googling newsboat and dropbox", and my hope would be that if someone else comes upon this and needs their hand held like I do, that person might also be helped by your response.

**Comment by evantravers** [comment]
@s1011205 I can't really tell from here what you did wrong… and honestly I've been not using Newsboat for a bit. :P

Newsboat needs to know where the `config`, `urls`, and `cache.db` are. They need to be located in a folder that it normally searches. `~/.newsboat` is a fine location… and you only need one folder I believe. XDG splits config and data, but dotdir is messy and puts it all in one… which is good for your case I think.

I can't tell from the commands you posted… but I'd try this:

1. Make one folder in dropbox, let's say it's called `~/Dropbox/sync/newsboat`. It has all the files in it.
2. remove any existing dotdir with `rm -rf ~/.newsboat`
3. symlink the dropbox folder -> the dotdir with `ln -s ~/Dropbox/sync/newsboat ~/.newsboat`

When you launch newsboat, it _should_ look first in the dotdir for the files it needs and be ok. Make sure that you are symlinking to the folder, not _inside_ the folder if that makes sense.

**Comment by evantravers** [comment]
(addendum: I haven't played with dropbox in a while. It wouldn't surprise me with all the clever things they've been doing recently that symlinks may not work.)

**Comment by s1011205** [comment]
Evan,

Thank you so much for your input and help. It turns out that your "Make sure that you are symlinking to the folder, not inside the folder if that makes sense" was spot on. What I initially did created a symlink folder inside the ~/.newsboat folder. By deleting the '~/.newsboat' file and creating a symlink ~/.newsboat folder, it worked. Thank you SO much for your help!

I tested newsboat by getting rid of the cache.db file and the 'newsboat' folder that contained it that was in ~/.local/share/, which just left the urls, the config file, and the other cache.db file, and newsboat worked fine without it. It worked both in regards to the feeds and in regard to remembering if I opened up a post entry. Somehow I had received the impression that newsboat might have had the one cache.db file in the ~/.newsboat folder and then had a second cache.db folder in a ~/.local/share/newsboat folder. That impression was wrong. It only needs the one. Now I can have just the one cache.db file, the urls file, and the config file in the one Dropbox folder, with a symlink ~/.newsboat folder pointing to it. This is great. If I was not in significant debt, I would want to tip you, but for now, you have my gratitude. Thank you for giving your time to and helping a total stranger, Evan.
